aws/subscriber.go

```diff
@@ -13,21 +12,33 @@ import (
 	"github.com/aws/aws-sdk-go/service/sqs"
 	"github.com/aws/aws-sdk-go/service/sqs/sqsiface"
 	pubsub "github.com/infobloxopen/atlas-pubsub"
+	"github.com/sirupsen/logrus"
 )
 
 // ErrAckDeadlineOutOfRange is returned whenever an invalid duration is passed to ExtendAckDeadline
 var ErrAckDeadlineOutOfRange = errors.New("The visibility timeout value is out of range. Values can be 0 to 43200 seconds")
 
+type SubscriberOption func(*awsSubscriber)
+
+func SubscribeWithLogger(logger *logrus.Logger) SubscriberOption {
+	return func(sub *awsSubscriber) {
+		sub.logger = logger
+	}
+}
+
 // NewSubscriber creates an AWS message broker that will subscribe to
 // the given topic with at-least-once message delivery semantics for the given
 // subscriptionID
 // TODO: info on permissions needed within the config to make this work
-func NewSubscriber(sess *session.Session, topic, subscriptionID string) (pubsub.Subscriber, error) {
-	return newSubscriber(sns.New(sess), sqs.New(sess), topic, subscriptionID)
+func NewSubscriber(sess *session.Session, topic, subscriptionID string, opts ...SubscriberOption) (pubsub.Subscriber, error) {
+	return newSubscriber(sns.New(sess), sqs.New(sess), topic, subscriptionID, opts...)
 }
 
-func newSubscriber(snsClient snsiface.SNSAPI, sqsClient sqsiface.SQSAPI, topic, subscriptionID string) (*awsSubscriber, error) {
-	subscriber := awsSubscriber{sns: snsClient, sqs: sqsClient, topic: topic, subscriptionID: subscriptionID}
+func newSubscriber(snsClient snsiface.SNSAPI, sqsClient sqsiface.SQSAPI, topic, subscriptionID string, opts ...SubscriberOption) (*awsSubscriber, error) {
+	subscriber := awsSubscriber{sns: snsClient, sqs: sqsClient, topic: topic, subscriptionID: subscriptionID, logger: logrus.StandardLogger()}
```

Comment:
I see this logger will be overrided, but maybe it will be better to use toolkit logger with some default level (logrus.InfoLevel) as a default logger? It up to you, for me it seems more consistent